code/game/machinery/kitchen/microwave.dm

```diff
@@ -346,61 +347,38 @@
 			return 1
 	return 0
 
-/obj/machinery/microwave/proc/start()
-	src.visible_message("<span class='notice'>The microwave turns on.</span>", "<span class='notice'>You hear a microwave.</span>")
-	src.operating = 1
-	src.icon_state = "mw1"
-	src.updateUsrDialog()
+/obj/machinery/microwave/proc/start(var/muck = FALSE)
+	visible_message("<span class='notice'>The microwave turns on.</span>", "<span class='notice'>You hear a microwave.</span>")
+	operating = TRUE
+	if(muck)
+		playsound(loc, 'sound/effects/splat.ogg', 50, 1) // Play a splat sound
+		icon_state = "mwbloody1" // Make it look dirty!!
+	else
+		icon_state = "mw1"
 	set_light(1.5)
 	soundloop.start()
 
-/obj/machinery/microwave/proc/abort()
-	after_finish_loop()
-	src.operating = 0 // Turn it off again aferwards
-	src.icon_state = "mw"
-	src.updateUsrDialog()
-
-/obj/machinery/microwave/proc/stop()
+/obj/machinery/microwave/proc/stop(var/muck = FALSE, var/broken = FALSE)
 	after_finish_loop()
-	src.operating = 0 // Turn it off again aferwards
-	src.icon_state = "mw"
-	src.updateUsrDialog()
-
-/obj/machinery/microwave/proc/dispose(var/message = 1)
-	for (var/atom/movable/A in contents)
-		A.forceMove(loc)
-	if (src.reagents.total_volume)
-		src.dirty++
-	src.reagents.clear_reagents()
-	if (message)
-		to_chat(usr, "<span class='notice'>You dispose of the microwave contents.</span>")
-	src.updateUsrDialog()
-
-/obj/machinery/microwave/proc/muck_start()
-	playsound(src.loc, 'sound/effects/splat.ogg', 50, 1) // Play a splat sound
-	src.icon_state = "mwbloody1" // Make it look dirty!!
-
-/obj/machinery/microwave/proc/muck_finish()
-	after_finish_loop()
-	src.visible_message("<span class='warning'>The microwave gets covered in muck!</span>")
-	src.dirty = 100 // Make it dirty so it can't be used util cleaned
-	src.flags = null //So you can't add condiments
-	src.icon_state = "mwbloody" // Make it look dirty too
-	src.operating = 0 // Turn it off again aferwards
-	src.updateUsrDialog()
-
-/obj/machinery/microwave/proc/broke()
-	after_finish_loop()
-	spark(src, 2, alldirs)
-	src.icon_state = "mwb" // Make it look all busted up and shit
-	src.visible_message("<span class='warning'>The microwave breaks!</span>") //Let them know they're stupid
-	src.broken = 2 // Make it broken so it can't be used util fixed
-	src.flags = null //So you can't add condiments
-	src.operating = 0 // Turn it off again aferwards
-	src.updateUsrDialog()
+	operating = FALSE // Turn it off again aferwards
+	if(muck || broken)
+		flags = null //So you can't add condiments
+		close_cooking_uis() // Can't use this anymore!

```

Comment:
This should not be needed as VueUI code should handle things that are broken and unusable via Topic states.